### ISSUE TITLE
Add the request_id to the response header

### ISSUE THREAD

**Issue by HMaker** [jrobichaud/django-structlog#231]
`django-structlog` makes it easy to process application logs and track issues in a per-request basis, it would be good if the library supported returning the actual `request_id` as a HTTP Response header like `X-Request-ID`, this way application users could include that ID in the support ticket and the maintainers could search the logs by the request ID to find the possible causes of that issue.

**Comment by jrobichaud** [comment]
Have you considered passed in a X-Correlation-ID in the request instead? 

https://django-structlog.readthedocs.io/en/latest/events.html

![image](https://user-images.githubusercontent.com/12142702/233801941-9070599a-30b1-412c-b35f-48a0c1c2af7a.png)

**Comment by HMaker** [comment]
In my case I use DRF to provide an API to end users, so I would like to just return the request ID in the HTTP headers.

**Comment by jrobichaud** [comment]
https://django-structlog.readthedocs.io/en/latest/api_documentation.html

Maybe you could do something with 

django_structlog.signals.bind_extra_request_finished_metadata

django_structlog.signals.bind_extra_request_failed_metadata

Get the request ID from the context and add it to the response.

**Comment by jrobichaud** [comment]
I just realized this signal is not called in case of an exception. 

Maybe we would need to change something for this.

**Comment by jrobichaud** [comment]
X-Request-ID id a non-official request header. I am not going to add it directly to `django-structlog` but I am willing to make it possible for you to do it with signals.

**Comment by jrobichaud** [comment]
I just released 5.1.0 with a change that will allow you to do what you want.

```python
from django.dispatch import receiver
from django_structlog import signals
import structlog


@receiver(signals.update_failure_response)
@receiver(signals.bind_extra_request_finished_metadata)
def add_request_id_to_error_response(response, logger, **kwargs):
    context = structlog.contextvars.get_merged_contextvars(logger)
    response['X-Request-ID'] = context["request_id"]
```

```bash
$ curl -vvv http://127.0.0.1:8001/async_view
*   Trying 127.0.0.1:8001...
* Connected to 127.0.0.1 (127.0.0.1) port 8001 (#0)
> GET /async_view HTTP/1.1
> Host: 127.0.0.1:8001
> User-Agent: curl/7.87.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< date: Sun, 23 Apr 2023 02:50:45 GMT
< server: uvicorn
< Content-Type: text/html; charset=utf-8
< X-Request-ID: 1a06b191-dfa0-4b6f-b645-fc3b108d2ae3
< X-Frame-Options: DENY
< Content-Length: 0
< Vary: Cookie
< X-Content-Type-Options: nosniff
< Referrer-Policy: same-origin
< Cross-Origin-Opener-Policy: same-origin
<
* Connection #0 to host 127.0.0.1 left intact
```

**Comment by HMaker** [comment]
Thanks, that's enough.
